routing/routing_helpers.cpp

```diff
@@ -18,6 +20,84 @@ namespace routing
 using namespace std;
 using namespace traffic;
 
+void FillSegmentInfo(vector<Segment> const & segments, vector<Junction> const & junctions,
+                     Route::TTurns const & turns, Route::TStreets const & streets,
+                     Route::TTimes const & times, shared_ptr<TrafficStash> const & trafficStash,
+                     vector<RouteSegment> & routeSegment)
+{
+  CHECK_EQUAL(segments.size() + 1, junctions.size(), ());
+  CHECK(!turns.empty(), ());
+  CHECK(!times.empty(), ());
+
+  CHECK(std::is_sorted(turns.cbegin(), turns.cend(), my::LessBy(&turns::TurnItem::m_index)), ());
+  CHECK(std::is_sorted(streets.cbegin(), streets.cend(), my::LessBy(&Route::TStreetItem::first)), ());
+  CHECK(std::is_sorted(times.cbegin(), times.cend(), my::LessBy(&Route::TTimeItem::first)), ());
+
+  CHECK_LESS(turns.back().m_index, junctions.size(), ());
+  if (!streets.empty())
+    CHECK_LESS(streets.back().first, junctions.size(), ());
+  CHECK_LESS(times.back().first, junctions.size(), ());
+
+  routeSegment.clear();
+  if (segments.empty())
+    return;
+
+  routeSegment.reserve(segments.size());
+  // Note. |turns|, |streets| and |times| have point based index.
+  // On the other hand turns, street names and times are considered for further end of |segments| after conversion
+  // to |segmentInfo|. It means that street, turn and time information of zero point will be lost after
+  // conversion to |segmentInfo|.
+  size_t turnIdx = turns[0].m_index != 0 ? 0 : 1;
+  size_t streetIdx = (!streets.empty() && streets[0].first != 0) ? 0 : 1;
+  size_t timeIdx = times[0].first != 0 ? 0 : 1;
+  double routeLengthMeters = 0.0;
+  double routeLengthMerc = 0.0;
+  double timeFromBeginningS = 0.0;
+
+  for (size_t i = 0; i < segments.size(); ++i)
+  {
+    size_t const segmentEndPointIdx = i + 1;
+
+    turns::TurnItem curTurn;
+    CHECK_LESS_OR_EQUAL(turnIdx, turns.size(), ());
+    if (turnIdx != turns.size() && turns[turnIdx].m_index == segmentEndPointIdx)
```

Comment:
It could be replace by "route". Or even better by "route polyline". "polyline" means Route::m_poly. It's a instance of FollowedPolyline. This instance contain all point of the route.